### PR TITLE
kubelet: convert node shutdown manager to contextual logging

### PR DESCRIPTION
**Pull request by pohly**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

A specific advantage of this change is that output checking becomes simpler. It
had race conditions in the
past (https://github.com/kubernetes/kubernetes/issues/108040). Now the test
works without modifying global state and can run in parallel to others.

#### Which issue(s) this PR fixes:
Related-to: https://github.com/kubernetes/klog/issues/317

#### Special notes for your reviewer:

This is the first PR which uses the new klog.ktesting for contextual logging in a unit test, so that code gets pulled into vendor.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux.go
Comment (pohly) [reply pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux.go:136]: By checking for the nil pointer here we can later be sure that manager.logger is usable. I'll add that as comment.

File: pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux.go
Comment (pohly) [reply pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux.go:135]: We can also decide to make Config.Logger a value and require all callers to set it. I think I already did that for all in-tree code.

Comment (pohly) [reply pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux.go]: I was too cautious. Requiring `Config.Logger` to be a valid value is simpler and doable here because this is an internal API where we can and want to change all callers.

Updated...

File: vendor/k8s.io/klog/v2/ktesting/testinglogger.go
Comment (pohly) [inline vendor/k8s.io/klog/v2/ktesting/testinglogger.go:42]: This looked like one of those cases where having some experimental code in klog is not going to cause problems because it is explicitly designed to be used only by unit tests.

If some third-party dependency of Kubernetes uses this and needs some other API than the one that gets used by Kubernetes, then all non-test code of that dependency will still work and the test code won't matter.

Having said that, we'll probably move this to non-experimental once we have used it a bit more in Kubernetes 1.25. See https://github.com/kubernetes/klog/issues/335

File: pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux_test.go
Comment (serathius) [inline pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux_test.go:689]: This is new. Was it intentional?

File: pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux_test.go
Comment (serathius) [inline pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux_test.go:692]: ditto

**Comment by pohly** [comment]
/assign @klueska  @dims 

For approval.

**Comment by dims** [comment]
happy to approve if we get a review from sig-node folks~

**Review by mrunalp — APPROVED** [review]

**Pull request merged**
